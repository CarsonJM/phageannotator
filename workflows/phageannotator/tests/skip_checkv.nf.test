nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"
    tag "workflows"
    tag "phageannotator"
    tag "phageannotator_skip_checkv"


    test("Parameters: skip_checkv = true") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists:true),
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ],
                    [
                        [ id:'test2' ],
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists:true)
                    ]
                )
                """
            }
            params {
                outdir                  = "$outputDir"
                skip_checkv             = true
                mash_screen_min_score   = 0.5
                genomad_min_score       = 0.01
                genomad_max_fdr         = 1
                genomad_disable_nn      = true
                genomad_sensitivity     = 0.1
                checkv_minimal_db       = true
                iphop_partial_test      = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert path("${outputDir}/VirusLifestyle/bacphlip/all_samples_anicluster_representatives.fna.bacphlip").readLines().size() == 5 },
                { assert path("${outputDir}/VirusAbundance/coverm/contig/all_samples_alignment_results.tsv").readLines().size() == 5 }
            )
        }
    }
}
