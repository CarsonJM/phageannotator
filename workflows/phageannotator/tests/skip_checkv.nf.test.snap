{
    "Parameters: skip_checkv = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.viromeqc.tsv:md5,728ab79cbb6c7afd0dc3f38c5ddfe642"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,b55f2c77b63adc41da980568819d1aae"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus_summary.tsv:md5,617215f2a883d4ec8d95aacd0b09d8a1"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus_summary.tsv:md5,f5b59da727b9793461f8b85cf7c818df"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus.fna.gz:md5,3221973668db26d28b3fcf3113d03b37"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus.fna.gz:md5,eb6c2478d8f4bb739ebaa322b591ec9a"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,57407ab897197c800302f101710832c1"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_taxonomy.tsv:md5,fecbf980c5e957d168c1141ce1b1769e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,b4c05c312e5bf4a3e01e0fa599472982"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,829d648756f80789738e5be6328c699f"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,829d648756f80789738e5be6328c699f"
                    ]
                ],
                "9": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,00a12ed45ed179a55e30c05dbb3d31f3",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,2b068b9678f63d6bb06a2a3652e92ba9",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,c0598196e761dbcdc90ff95a2f623b9c",
                    "versions.yml:md5,c97cb7740ce1aa3cf22cb6e8be2d62ce",
                    "versions.yml:md5,d73a3ecea014f00c8d95513555c42d23",
                    "versions.yml:md5,de688faac0a9124c4f2db74ce1b2c795",
                    "versions.yml:md5,dfa7110695181c4e8daa47d384563588",
                    "versions.yml:md5,f383a8d4db50b157e4a6a0ef4670fbc6"
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,57407ab897197c800302f101710832c1"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus.fna.gz:md5,3221973668db26d28b3fcf3113d03b37"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus.fna.gz:md5,eb6c2478d8f4bb739ebaa322b591ec9a"
                    ]
                ],
                "host_predictions_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "instrain_gene_info": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,829d648756f80789738e5be6328c699f"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,829d648756f80789738e5be6328c699f"
                    ]
                ],
                "marker_taxonomy_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_taxonomy.tsv:md5,fecbf980c5e957d168c1141ce1b1769e"
                    ]
                ],
                "pharokka_output_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_cds_final_merged_output.tsv:md5,b4c05c312e5bf4a3e01e0fa599472982"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,005aa3fca95c916b586f09b1c12aae6f",
                    "versions.yml:md5,00a12ed45ed179a55e30c05dbb3d31f3",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,0c310e8a973612ff99e0df847fb656ea",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,2b068b9678f63d6bb06a2a3652e92ba9",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,374c63bf7c31c6c34acbf8d74e6c7dcc",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,5bf636608598d8b495d8b38fcee37dd0",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,716410d7e9dc048a64ada71202cbc23e",
                    "versions.yml:md5,74b2068d24b17a2f4a097a3063da3dae",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,afc743a69ab04322fbc732b24e18d208",
                    "versions.yml:md5,b6a0d868f5f2be1093be05fc0638008d",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,bcf62c8c784b9f46c8372e86654cae6b",
                    "versions.yml:md5,c0598196e761dbcdc90ff95a2f623b9c",
                    "versions.yml:md5,c97cb7740ce1aa3cf22cb6e8be2d62ce",
                    "versions.yml:md5,d73a3ecea014f00c8d95513555c42d23",
                    "versions.yml:md5,de688faac0a9124c4f2db74ce1b2c795",
                    "versions.yml:md5,dfa7110695181c4e8daa47d384563588",
                    "versions.yml:md5,f383a8d4db50b157e4a6a0ef4670fbc6"
                ],
                "virus_classification_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.fasta_w_screen_hits_virus_summary.tsv:md5,617215f2a883d4ec8d95aacd0b09d8a1"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits_virus_summary.tsv:md5,f5b59da727b9793461f8b85cf7c818df"
                    ]
                ],
                "virus_enrichment_tsv": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.viromeqc.tsv:md5,728ab79cbb6c7afd0dc3f38c5ddfe642"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,b55f2c77b63adc41da980568819d1aae"
                    ]
                ],
                "virus_quality_tsv": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-15T15:44:02.602024455"
    }
}